### ISSUE TITLE
[<Ray component: Core] dataset.sum crashed: AttributeError: 'pyarrow.lib.Table' object has no attribute 'reset_index' 

### ISSUE THREAD

**Issue by Ox0400** [ray-project/ray#39155]
### What happened + What you expected to happen

Ray data dataset `sum()` crashed. Expect result is a `dict` or `int`.
Some times dataset format is `pyarrow`.

```python
# I add some debug log on my local for /usr/local/lib/python3.10/site-packages/ray/data/_internal/pandas_block.py
# e.g
... 
class PandasBlockAccessor(TableBlockAccessor):
    ROW_TYPE = PandasRow

    def __init__(self, table: "pandas.DataFrame"):
        super().__init__(table)
        import pyarrow
        if isinstance(self._table, pyarrow.lib.Table):
            print('init table ', type(self._table), self._table)
...
    def slice(self, start: int, end: int, copy: bool = False) -> "pandas.DataFrame":
        print('call slice self._table', type(self._table), self._table)
        view = self._table[start:end]
        view.reset_index(drop=True, inplace=True)
        if copy:
            view = view.copy(deep=True)
        return view
...

```

```txt
2023-08-31 18:50:36,739 INFO worker.py:1431 -- Connecting to existing Ray cluster at address: 172.18.0.7:6379...
2023-08-31 18:50:36,755 INFO worker.py:1612 -- Connected to Ray cluster. View the dashboard at 172.18.0.7:8265 
args split word is ['\n']
2023-08-31 18:50:38,073 INFO read_api.py:374 -- To satisfy the requested parallelism of 8, each read task output will be split into 8 smaller blocks.
Filtering <256 and wd...: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 2312.19it/s]
2023-08-31 18:50:38,211 INFO streaming_executor.py:92 -- Executing DAG InputDataBuffer[Input] -> TaskPoolMapOperator[ReadJSON->SplitBlocks(8)] -> TaskPoolMapOperator[Filter(<lambda>)]
2023-08-31 18:50:38,213 INFO streaming_executor.py:93 -- Execution config: ExecutionOptions(resource_limits=ExecutionResources(cpu=3, gpu=None, object_store_memory=None), locality_with_output=False, preserve_order=False, actor_locality_enabled=True, verbose_progress=True)
Hashing...:   0%|                                                                                                                                                            | 0/1 [00:00<?, ?it/s]2023-08-31 18:50:40,125 INFO streaming_executor.py:92 -- Executing DAG InputDataBuffer[Input] -> TaskPoolMapOperator[ReadJSON->SplitBlocks(8)] -> TaskPoolMapOperator[Filter(<lambda>)->MapBatches(compute_hashes)]                                                                                                                                                                                   
2023-08-31 18:50:40,125 INFO streaming_executor.py:93 -- Execution config: ExecutionOptions(resource_limits=ExecutionResources(cpu=3, gpu=None, object_store_memory=None), locality_with_output=False, preserve_order=False, actor_locality_enabled=True, verbose_progress=True)
Hashing...: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  7.19it/s]
total lines count 29                                                                                                                                                                               
Deduping...: 15it [00:00, 309.63it/s]                                                                                                                                                              
Deduping...: 0it [00:00, ?it/s]
remove lines count:  2
Filtering...:   0%|                                                                                                                                                          | 0/1 [00:00<?, ?it/s]2023-08-31 18:50:40,313 INFO dataset.py:2180 -- Tip: Use `take_batch()` instead of `take() / show()` to return records in pandas or numpy batch format.
2023-08-31 18:50:40,315 INFO streaming_executor.py:92 -- Executing DAG InputDataBuffer[Input] -> TaskPoolMapOperator[ReadJSON->SplitBlocks(8)] -> TaskPoolMapOperator[Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>)] -> AllToAllOperator[Aggregate]
2023-08-31 18:50:40,316 INFO streaming_executor.py:93 -- Execution config: ExecutionOptions(resource_limits=ExecutionResources(cpu=3, gpu=None, object_store_memory=None), locality_with_output=False, preserve_order=False, actor_locality_enabled=True, verbose_progress=True)
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21517) call slice self._table <class 'pandas.core.frame.DataFrame'>                                                                 
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21516) call slice self._table <class 'pandas.core.frame.DataFrame'>                                                                 
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   simhash  ... after_lines                                      
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21566) 0    None  ...           0                                                                                                   
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21566) 1    None  ...           2                                                                                                   
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21566)                                                                                                                              
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21566) [2 rows x 9 columns]                                                                                                         
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21517)   simhash  ... after_lines                                                                                                   
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21517) 0    None  ...           2                                                                                                   
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21517)                                                                                                                              
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21517) [1 rows x 9 columns]                                                                                                         
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21516)   simhash  ... after_lines                                                                                                   
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21516) 0    None  ...           4                                                                                                   
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21516) 1    None  ...           5                                                                                                   
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21516)                                                                                                                              
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21516) [2 rows x 9 columns]                                                                                                         
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21517) call slice self._table <class 'pandas.core.frame.DataFrame'>                                                                 
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21516) call slice self._table <class 'pandas.core.frame.DataFrame'>   simhash  ... after_lines                                      
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21516) 0    None  ...           0                                                                                                   
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21516) 1    None  ...           1                                                                                                   
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21516)                                                                                                                              
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21516) [2 rows x 9 columns]                                                                                                         
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   simhash  ... after_lines                                      
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21566) 0    None  ...           1                                                                                                   
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21566)                                                                                                                              
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21566) [1 rows x 9 columns]                                                                                                         
                                                                                                                                                                                                  (Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>                                                                  
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21517)   simhash  ... after_lines                                                                                                   
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21517) 0    None  ...           4                                                                                                   
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21517)                                                                                                                              
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21517) [1 rows x 9 columns]                                                                                                         
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21566)   simhash                                     txt  ... before_lines after_lines                                              
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21566) 0    None  延吉路低楼层满五年看房方便可随时。\n我已阅读并同意《利发国际用户使用协议》  ...            2           2          
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21566)                                                                                                                              
(Filter(<lambda>)->MapBatches(dedup)->MapBatches(<lambda>) pid=21566) [1 rows x 9 columns]                                                                                                         
(reduce pid=21566) init table  <class 'pyarrow.lib.Table'> pyarrow.Table                                                                                                                           
(reduce pid=21566) sum(before_lines): list<item: int64>                                                                                                                                            
(reduce pid=21566)   child 0, item: int64                                                                                                                                                          
(reduce pid=21566) sum(after_lines): list<item: int64>                                                                                                                                             
(reduce pid=21566)   child 0, item: int64                                                                                                                                                          
(reduce pid=21566) ----                                                                                                                                                                            
(reduce pid=21566) sum(before_lines): [[[0,0]]]                                                                                                                                                    
(reduce pid=21566) sum(after_lines): [[[0,0]]]                                                                                                                                                     
(reduce pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   sum(before_lines) sum(after_lines)                                                                               
(reduce pid=21566) 0            [7, 1]           [2, 1]                                                                                                                                            
(reduce pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   sum(before_lines) sum(after_lines)                                                                               
(reduce pid=21566) 0            [2, 1]           [2, 1]                                                                                                                                            
(reduce pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   sum(before_lines) sum(after_lines)                                                                               
(reduce pid=21566) 0            [9, 1]           [9, 1]                                                                                                                                            
(reduce pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   sum(before_lines) sum(after_lines)                                                                               
(reduce pid=21566) 0            [4, 1]           [1, 1]                                                                                                                                            
(reduce pid=21566) call slice self._table <class 'pyarrow.lib.Table'> pyarrow.Table                                                                                                                
(reduce pid=21566) sum(before_lines): list<item: int64>                                                                                                                                            
(reduce pid=21566)   child 0, item: int64                                                                                                                                                          
(reduce pid=21566) sum(after_lines): list<item: int64>                                                                                                                                             
(reduce pid=21566)   child 0, item: int64                                                                                                                                                          
(reduce pid=21566) ----                                                                                                                                                                            
(reduce pid=21566) sum(before_lines): [[[0,0]]]                                                                                                                                                    
(reduce pid=21566) sum(after_lines): [[[0,0]]]                                                                                                                                                     
(reduce pid=21566) init table  <class 'pyarrow.lib.Table'> pyarrow.Table                                                                                                                           
(reduce pid=21566) sum(before_lines): list<item: int64>                                                                                                                                            
(reduce pid=21566)   child 0, item: int64                                                                                                                                                          
(reduce pid=21566) sum(after_lines): list<item: int64>                                                                                                                                             
(reduce pid=21566)   child 0, item: int64                                                                                                                                                          
(reduce pid=21566) ----                                                                                                                                                                            
(reduce pid=21566) sum(before_lines): [[[0,0]]]                                                                                                                                                    
(reduce pid=21566) sum(after_lines): [[[0,0]]]                                                                                                                                                     
(reduce pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   sum(before_lines) sum(after_lines)                                                                               
(reduce pid=21566) 0            [7, 1]           [2, 1]                                                                                                                                            
(reduce pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   sum(before_lines) sum(after_lines)                                                                               
(reduce pid=21566) 0            [2, 1]           [2, 1]                                                                                                                                            
(reduce pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   sum(before_lines) sum(after_lines)                                                                               
(reduce pid=21566) 0            [9, 1]           [9, 1]                                                                                                                                            
(reduce pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   sum(before_lines) sum(after_lines)                                                                               
(reduce pid=21566) 0            [4, 1]           [1, 1]                                                                                                                                            
(reduce pid=21566) call slice self._table <class 'pyarrow.lib.Table'> pyarrow.Table                                                                                                                
(reduce pid=21566) sum(before_lines): list<item: int64>                                                                                                                                            
(reduce pid=21566)   child 0, item: int64                                                                                                                                                          
(reduce pid=21566) sum(after_lines): list<item: int64>                                                                                                                                             
(reduce pid=21566)   child 0, item: int64                                                                                                                                                          
(reduce pid=21566) ----                                                                                                                                                                            
(reduce pid=21566) sum(before_lines): [[[0,0]]]                                                                                                                                                    
(reduce pid=21566) sum(after_lines): [[[0,0]]]                                                                                                                                                     
(reduce pid=21566) init table  <class 'pyarrow.lib.Table'> pyarrow.Table                                                                                                                           
(reduce pid=21566) sum(before_lines): list<item: int64>                                                                                                                                            
(reduce pid=21566)   child 0, item: int64                                                                                                                                                          
(reduce pid=21566) sum(after_lines): list<item: int64>                                                                                                                                             
(reduce pid=21566)   child 0, item: int64                                                                                                                                                          
(reduce pid=21566) ----                                                                                                                                                                            
(reduce pid=21566) sum(before_lines): [[[0,0]]]                                                                                                                                                    
(reduce pid=21566) sum(after_lines): [[[0,0]]]                                                                                                                                                     
(reduce pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   sum(before_lines) sum(after_lines)                                                                               
(reduce pid=21566) 0            [7, 1]           [2, 1]                                                                                                                                            
(reduce pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   sum(before_lines) sum(after_lines)                                                                               
(reduce pid=21566) 0            [2, 1]           [2, 1]                                                                                                                                            
(reduce pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   sum(before_lines) sum(after_lines)                                                                               
(reduce pid=21566) 0            [9, 1]           [9, 1]                                                                                                                                            
(reduce pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   sum(before_lines) sum(after_lines)                                                                               
(reduce pid=21566) 0            [4, 1]           [1, 1]                                                                                                                                            
(reduce pid=21566) call slice self._table <class 'pyarrow.lib.Table'> pyarrow.Table                                                                                                                
(reduce pid=21566) sum(before_lines): list<item: int64>                                                                                                                                            
(reduce pid=21566)   child 0, item: int64                                                                                                                                                          
(reduce pid=21566) sum(after_lines): list<item: int64>                                                                                                                                             
(reduce pid=21566)   child 0, item: int64                                                                                                                                                          
(reduce pid=21566) ----                                                                                                                                                                            
(reduce pid=21566) sum(before_lines): [[[0,0]]]                                                                                                                                                    
(reduce pid=21566) sum(after_lines): [[[0,0]]]                                                                                                                                                     
Filtering...:   0%|                                                                                                                                                          | 0/1 [00:03<?, ?it/s]
Traceback (most recent call last):                                                                                                                                                                 
  File "/var/work/ccnet1.py", line 213, in <module>                                                                                                                                                
    with timer("Total"):                                                                                                                                                                           
  File "/usr/local/lib/python3.10/site-packages/text_dedup/utils/timer.py", line 19, in __exit__                                                                                                   
    raise exc_val                                                                                                                                                                                  
  File "/var/work/ccnet1.py", line 298, in <module>
    with timer("Filtering"):
  File "/usr/local/lib/python3.10/site-packages/text_dedup/utils/timer.py", line 19, in __exit__
    raise exc_val
  File "/var/work/ccnet1.py", line 310, in <module>
    lines_diff = ds.select_columns(['before_lines', 'after_lines']).sum(['before_lines', 'after_lines'])
  File "/usr/local/lib/python3.10/site-packages/ray/data/dataset.py", line 1837, in sum
    ret = self._aggregate_on(Sum, on, ignore_nulls)
  File "/usr/local/lib/python3.10/site-packages/ray/data/dataset.py", line 4457, in _aggregate_on
    return self.aggregate(*aggs)
  File "/usr/local/lib/python3.10/site-packages/ray/data/dataset.py", line 1796, in aggregate
    ret = self.groupby(None).aggregate(*aggs).take(1)
  File "/usr/local/lib/python3.10/site-packages/ray/data/dataset.py", line 2185, in take
    for row in self.iter_rows():
  File "/usr/local/lib/python3.10/site-packages/ray/data/iterator.py", line 237, in iter_rows
    for batch in self.iter_batches(**iter_batch_args):
  File "/usr/local/lib/python3.10/site-packages/ray/data/iterator.py", line 159, in iter_batches
    block_iterator, stats, blocks_owned_by_consumer = self._to_block_iterator()
  File "/usr/local/lib/python3.10/site-packages/ray/data/_internal/iterator/iterator_impl.py", line 32, in _to_block_iterator
    block_iterator, stats, executor = ds._plan.execute_to_iterator()
  File "/usr/local/lib/python3.10/site-packages/ray/data/_internal/plan.py", line 538, in execute_to_iterator
    block_iter = itertools.chain([next(gen)], gen)
  File "/usr/local/lib/python3.10/site-packages/ray/data/_internal/execution/legacy_compat.py", line 51, in execute_to_legacy_block_iterator
    for bundle in bundle_iter:
  File "/usr/local/lib/python3.10/site-packages/ray/data/_internal/execution/interfaces.py", line 548, in __next__
    return self.get_next()
  File "/usr/local/lib/python3.10/site-packages/ray/data/_internal/execution/streaming_executor.py", line 129, in get_next
    raise item
  File "/usr/local/lib/python3.10/site-packages/ray/data/_internal/execution/streaming_executor.py", line 187, in run
    while self._scheduling_loop_step(self._topology) and not self._shutdown:
  File "/usr/local/lib/python3.10/site-packages/ray/data/_internal/execution/streaming_executor.py", line 267, in _scheduling_loop_step
    update_operator_states(topology)
  File "/usr/local/lib/python3.10/site-packages/ray/data/_internal/execution/streaming_executor_state.py", line 359, in update_operator_states
    op.all_inputs_done()
  File "/usr/local/lib/python3.10/site-packages/ray/data/_internal/execution/operators/base_physical_operator.py", line 88, in all_inputs_done
    self._output_buffer, self._stats = self._bulk_fn(self._input_buffer, ctx)
  File "/usr/local/lib/python3.10/site-packages/ray/data/_internal/planner/aggregate.py", line 75, in fn
    return scheduler.execute(refs, num_outputs, ctx)
  File "/usr/local/lib/python3.10/site-packages/ray/data/_internal/planner/exchange/pull_based_shuffle_task_scheduler.py", line 89, in execute
    new_metadata = reduce_bar.fetch_until_complete(list(new_metadata))
  File "/usr/local/lib/python3.10/site-packages/ray/data/_internal/progress_bar.py", line 93, in fetch_until_complete
    for ref, result in zip(done, ray.get(done)):
  File "/usr/local/lib/python3.10/site-packages/ray/_private/auto_init_hook.py", line 24, in auto_init_wrapper
    return fn(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/ray/_private/client_mode_hook.py", line 103, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/ray/_private/worker.py", line 2524, in get
    raise value.as_instanceof_cause()
ray.exceptions.RayTaskError(AttributeError): ray::reduce() (pid=21566, ip=172.18.0.7)
  File "/usr/local/lib/python3.10/site-packages/ray/data/_internal/planner/exchange/aggregate_task_spec.py", line 71, in reduce
    return BlockAccessor.for_block(mapper_outputs[0]).aggregate_combined_blocks(
  File "/usr/local/lib/python3.10/site-packages/ray/data/_internal/pandas_block.py", line 521, in aggregate_combined_blocks
    next_row = next(iter)
  File "/usr/local/lib/python3.10/heapq.py", line 373, in merge
    value = next()
  File "/usr/local/lib/python3.10/site-packages/ray/data/_internal/table_block.py", line 194, in __next__
    row = outer._get_row(self._cur)
  File "/usr/local/lib/python3.10/site-packages/ray/data/_internal/table_block.py", line 159, in _get_row
    base_row = self.slice(index, index + 1, copy=copy)
  File "/usr/local/lib/python3.10/site-packages/ray/data/_internal/pandas_block.py", line 160, in slice
    view.reset_index(drop=True, inplace=True)
AttributeError: 'pyarrow.lib.Table' object has no attribute 'reset_index'
(reduce pid=21566) init table  <class 'pyarrow.lib.Table'> pyarrow.Table
(reduce pid=21566) sum(before_lines): list<item: int64>
(reduce pid=21566)   child 0, item: int64
(reduce pid=21566) sum(after_lines): list<item: int64>
(reduce pid=21566)   child 0, item: int64
(reduce pid=21566) ----
(reduce pid=21566) sum(before_lines): [[[0,0]]]
(reduce pid=21566) sum(after_lines): [[[0,0]]]
(reduce pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   sum(before_lines) sum(after_lines)
(reduce pid=21566) 0            [7, 1]           [2, 1]
(reduce pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   sum(before_lines) sum(after_lines)
(reduce pid=21566) 0            [2, 1]           [2, 1]
(reduce pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   sum(before_lines) sum(after_lines)
(reduce pid=21566) 0            [9, 1]           [9, 1]
(reduce pid=21566) call slice self._table <class 'pandas.core.frame.DataFrame'>   sum(before_lines) sum(after_lines)
(reduce pid=21566) 0            [4, 1]           [1, 1]
(reduce pid=21566) call slice self._table <class 'pyarrow.lib.Table'> pyarrow.Table
(reduce pid=21566) sum(before_lines): list<item: int64>
(reduce pid=21566)   child 0, item: int64
(reduce pid=21566) sum(after_lines): list<item: int64>
(reduce pid=21566)   child 0, item: int64
(reduce pid=21566) ----
(reduce pid=21566) sum(before_lines): [[[0,0]]]
(reduce pid=21566) sum(after_lines): [[[0,0]]]
root@fed1da28d58f:/var/work# 
```

### Versions / Dependencies

Python 3.10.12
pyarrow==13.0.0
ray==2.6.3

### Reproduction script

```python
import ray
ray.init('auto')
ray.data.DataContext.get_current().execution_options.verbose_progress = True
alive_cpu = int(ray.available_resources()['CPU'] * 0.95)

ctx = ray.data.DataContext.get_current()
ctx.execution_options.resource_limits.cpu = alive_cpu

def func(records):
    for i, _id in enumerate(records['id'])):
        records['tags'][i] = list(records['tags'][i])
    return records

ds = ray_ds.map_batches(func)
print (ds.sum(['id', 'score']) )

```

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by scottjlee** [comment]
Hi @Ox0400 , what is `ray_ds` in the reproducible above?

**Comment by Ox0400** [comment]
> Hi @Ox0400 , what is `ray_ds` in the reproducible above?

```
ds = ray.data.read_json('xxx')
ray_ds = ds.groupby('xxx').map_groups(xxx)
...
```
```

**Comment by andriiuber** [comment]
@scottjlee @anyscalesam Hi folks, what's the progress on this issue? Is it on track to be fixed in 2.10?
